frontend/src/app/Providers.tsx

```diff
@@ -2,11 +2,14 @@
 import React from "react";
 import { AuthProvider } from "@/contexts/AuthContext";
 import MessageReceiver from "@/app/MessageReceiver";
+import { ThemeProvider } from "@/contexts/ThemeContext";
 
 const Providers = ({ children }: { children: React.ReactNode }) => {
   return (
     <AuthProvider>
-      <MessageReceiver>{children}</MessageReceiver>
+      <ThemeProvider>
```

Comment:
With the current approach, when navigating to a subpage like `/board` or `/user-boards`, the page initially loads with the light theme, and only after a moment does it switch to the dark theme if that's what we've chosen. This flicker between themes doesn't seem ideal.

I came across a tutorial on implementing dark mode in Next.js without this flicker. It might be useful for you to check it out:  
[Next.js Dark Mode with No Flicker + Tailwind CSS](https://www.youtube.com/watch?v=7zqI4qMDdg8).

Reply:
Works like a charm!